lib/crowbar/client/app/upgrade.rb

```diff
@@ -0,0 +1,280 @@
+          the found rows.
+        LONGDESC
+
+        def services
+          Command::Upgrade::Services.new(
+            *command_params
+          ).execute
+        rescue => e
+          catch_errors(e)
+        end
+
+        desc "prechecks",
+          "Perform checks to make sure Crowbar can be upgraded"
+
+        long_desc <<-LONGDESC
+          `prechecks` will perform a sanity check on the Crowbar server to
+          make sure it can be upgraded.
+
+          With --format <format> option you can choose an output format
+          with the available options table, json or plain. You can also
+          use the shortcut options --table, --json or --plain.
+
+          With --filter <filter> option you can limit the result of
+          printed out rows. You can use any substring that is part of
+          the found rows.
+        LONGDESC
+
+        method_option :format,
+          type: :string,
+          default: "table",
+          banner: "<format>",
+          desc: "Format of the output, valid formats are table, json or plain"
+
+        method_option :table,
+          type: :boolean,
+          default: false,
+          aliases: [],
+          desc: "Format output as table, a shortcut for --format table option"
+
+        method_option :json,
+          type: :boolean,
+          default: false,
+          aliases: [],
+          desc: "Format output as table, a shortcut for --format json option"
+
+        method_option :plain,
+          type: :boolean,
+          default: false,
+          aliases: [],
+          desc: "Format output as table, a shortcut for --format plain option"
+
+        method_option :filter,
+          type: :string,
+          default: nil,
+          banner: "<filter>",
+          desc: "Filter by criteria, display only data that contains filter"
+
+        def prechecks
+          Command::Upgrade::Prechecks.new(
+            *command_params
+          ).execute
+        rescue => e
+          catch_errors(e)
+        end
+
+        desc "crowbar",
+          "Upgrade Crowbar"
+
+        long_desc <<-LONGDESC
+          `crowbar` will upgrade the operating system of the Crowbar server.
+        LONGDESC
+
+        def crowbar
+          Command::Upgrade::Crowbar.new(
+            *command_params
+          ).execute
+        rescue => e
+          catch_errors(e)
+        end
+
+        desc "node NODE",
+          "Upgrade a single node"
+
+        long_desc <<-LONGDESC
+          `node NODE` will upgrade the operating system of a single node.
+        LONGDESC
+
+        def node(node)
+          Command::Upgrade::Node.new(
+            *command_params(
+              node: node
+            )
+          ).execute
+        rescue => e
+          catch_errors(e)
+        end
+
+        desc "backup COMPONENT",
+          "Create a backup of a component (crowbar|openstack)"
+
+        long_desc <<-LONGDESC
+          `backup COMPONENT` will create a backup of Crowbar or the
```

Comment:
why? and btw this has nothing to do with `crowbar-client`, that would be a `crowbar-core` issue, so please open an issue there explaining the why